ci/test_cpp.sh

```diff
@@ -0,0 +1,101 @@
+#!/bin/bash
+# Copyright (c) 2022, NVIDIA CORPORATION.
+
+set -euo pipefail
+
+. /opt/conda/etc/profile.d/conda.sh
+
+rapids-logger "Generate C++ testing dependencies"
+rapids-dependency-file-generator \
+  --output conda \
+  --file_key test_cpp \
+  --matrix "cuda=${RAPIDS_CUDA_VERSION%.*};arch=$(arch)" | tee env.yaml
+
+rapids-mamba-retry env create --force -f env.yaml -n test
+
+# Temporarily allow unbound variables for conda activation.
+set +u
+conda activate test
+set -u
+
+CPP_CHANNEL=$(rapids-download-conda-from-s3 cpp)
+RAPIDS_TESTS_DIR=${RAPIDS_TESTS_DIR:-"${PWD}/test-results"}/
```

Comment:
I don’t like adding the slash here. This line should only handle the default value logic. I’d put the slash in the original environment variable definition and make this change:
```suggestion
RAPIDS_TESTS_DIR=${RAPIDS_TESTS_DIR:-"${PWD}/test-results/"}
```

Reply:
Or maybe add the slash in the places where it’s needed below? Then we can avoid the missing slash in `"${RAPIDS_TESTS_DIR}${test_name}.cs.log"`.

Reply:
I was torn on this...

> Or maybe add the slash in the places where it’s needed below?

The trailing slash is actually needed everywhere. Without the trailing slash, the C++ tests throw this error https://github.com/rapidsai/cudf/actions/runs/3586797596/jobs/6036781772#step:6:66377.

So I thought about putting the trailing slash in the shared workflow variable definition below, but then there's a discrepancy between the `RAPIDS_COVERAGE_DIR` and `RAPIDS_TESTS_DIR` variables (e.g. one has a trailing slash, but the other doesn't).

https://github.com/rapidsai/shared-action-workflows/blob/4bf344150bf50071678fe9d04774931ebb974f78/.github/workflows/conda-python-tests.yaml#L87-L88

I'm open to changing this if you want, but there are implications wherever we decide to move it.